Objects/unicodeobject.c

```diff
@@ -5043,6 +5090,145 @@ ascii_decode(const char *start, const char *end, Py_UCS1 *dest)
     return p - start;
 }
 
+#if (defined(__clang__) || defined(__GNUC__))
+#define HAS_CTZ 1
```

Comment:
Could we delegate this to autoconf instead? In addition, the intrinics seems to return an `int` but not an `unsigned int`: https://gcc.gnu.org/onlinedocs/gcc/Other-Builtins.html#index-_005f_005fbuiltin_005fctzll.

Reply:
we already use similar builtins by these macros.
I want to use same return value type with MSVC version.